src/__tests__/unit/controllers/static-router/test.tsx

```diff
@@ -116,6 +116,40 @@ describe('<StaticRouter />', () => {
       });
     });
 
+    it('should respect max wait time when fetching resources', async () => {
+      const data = await StaticRouter.requestResources({
+        // @ts-ignore
+        routes: mockedRoutes,
+        location: '/',
+        timeout: 350,
+      });
+
+      expect(data).toEqual({
+        BYE: {
+          key: {
+            data: null,
+            error: {
+              message: 'Resource timed out: BYE',
+              name: 'TimeoutError',
+              stack: expect.any(String),
+            },
+            loading: true,
+            promise: null,
+            expiresAt,
+          },
+        },
+        HI: {
+          key: {
+            data: 'hello world',
+            error: null,
+            loading: false,
+            promise: null,
+            expiresAt,
+          },
+        },
+      });
+    });
+
```

Comment:
Can we add another integration test: `should re-trigger requests for timed out resources when mounted` ?